src/System.Device.Gpio/System/Device/Gpio/GpioController.Linux.cs

```diff
@@ -30,6 +31,22 @@ public GpioController(PinNumberingScheme numberingScheme)
         /// <returns>A driver that works with the board the program is executing on.</returns>
         private static GpioDriver GetBestDriverForBoard()
         {
+            try
+            {
+                string modelInfo = File.ReadAllText(ModelInfoPath);
```

Comment:
This probably won't matter since as @krwq suggested we want to move this to IoT.Device.Bindings package for now, but in the future I think we don't want to have the logic of GetBestDriverForBoard to change for each new PCB we add. We only did it for the RPi because that is the most used board, and fall back to the Generic one but in reality what we would want is to keep it just like that, and then for each new PCB support we add we will require people to manually select this as the driver they use, like: `var controller = new GpioController(numberingScheme, new OrangePiZeroDriver());`

Reply:
I think we should eventually have some extensibility model there but I wouldn't add it yet until we have more boards because we don't have enough data to design APIs sanely yet